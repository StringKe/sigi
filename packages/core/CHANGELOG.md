# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.9.0](https://github.com/sigi-framework/sigi/compare/v2.8.6...v2.9.0) (2021-05-11)

### Bug Fixes

- dependencies list ([4a44cdd](https://github.com/sigi-framework/sigi/commit/4a44cdd65b1fa83ba949f6993c2708d2f6c2b722))

### Features

- **core:** ssr effect now could be marked retry on client ([116df59](https://github.com/sigi-framework/sigi/commit/116df5952c8f45cda94a0d352a5c538b78307ffe))

### Performance Improvements

- **core:** reduce js engine hidden path mismatch ([3224344](https://github.com/sigi-framework/sigi/commit/32243447ba72b7aa957cc52ef7460aaef118e175))
- **react:** fast code path on server ([f7df497](https://github.com/sigi-framework/sigi/commit/f7df49708808f5421f40798151108280ab669359))
- **ssr:** add fast path for ssr di logic ([8cd2f5a](https://github.com/sigi-framework/sigi/commit/8cd2f5a4517fed2f01058cc39d410af63fa53a3b))

## [2.8.6](https://github.com/sigi-framework/sigi/compare/v2.8.5...v2.8.6) (2021-05-06)

**Note:** Version bump only for package @sigi/core

## [2.8.5](https://github.com/sigi-framework/sigi/compare/v2.8.4...v2.8.5) (2021-04-28)

**Note:** Version bump only for package @sigi/core

## [2.8.2](https://github.com/sigi-framework/sigi/compare/v2.8.1...v2.8.2) (2021-03-29)

**Note:** Version bump only for package @sigi/core

# [2.8.0](https://github.com/sigi-framework/sigi/compare/v2.7.0...v2.8.0) (2021-02-22)

**Note:** Version bump only for package @sigi/core

## [2.5.1](https://github.com/sigi-framework/sigi/compare/v2.5.0...v2.5.1) (2021-01-22)

### Bug Fixes

- **core:** setup store immediately after set default state ([d5d2631](https://github.com/sigi-framework/sigi/commit/d5d2631898645a6eb938f7cb8ff5bacac8a9efe6))

## [2.4.5](https://github.com/sigi-framework/sigi/compare/v2.4.4...v2.4.5) (2020-12-14)

### Bug Fixes

- **core:** epic do not follow rxjs control ([e27ab82](https://github.com/sigi-framework/sigi/commit/e27ab824d9b99846654087b840c8b2cccea9af6c))

## [2.4.4](https://github.com/sigi-framework/sigi/compare/v2.4.3...v2.4.4) (2020-11-27)

**Note:** Version bump only for package @sigi/core

## [2.4.3](https://github.com/sigi-framework/sigi/compare/v2.4.2...v2.4.3) (2020-11-22)

**Note:** Version bump only for package @sigi/core

## [2.4.2](https://github.com/sigi-framework/sigi/compare/v2.4.1...v2.4.2) (2020-10-15)

**Note:** Version bump only for package @sigi/core

## [2.4.1](https://github.com/sigi-framework/sigi/compare/v2.4.0...v2.4.1) (2020-08-27)

**Note:** Version bump only for package @sigi/core

# [2.4.0](https://github.com/sigi-framework/sigi/compare/v2.3.1...v2.4.0) (2020-07-28)

### Features

- **core:** get rid of symbols ([6d7a848](https://github.com/sigi-framework/sigi/commit/6d7a84801e4ce2a7be731ad13e26e56cd0970135))
- **react:** expose common actions to dispatcher hooks ([20dd063](https://github.com/sigi-framework/sigi/commit/20dd0631d2501e4b912210fef1f2b69addadb612))

## [2.3.1](https://github.com/sigi-framework/sigi/compare/v2.3.0...v2.3.1) (2020-07-09)

**Note:** Version bump only for package @sigi/core

# [2.3.0](https://github.com/sigi-framework/sigi/compare/v2.2.0...v2.3.0) (2020-07-05)

**Note:** Version bump only for package @sigi/core

# [2.2.0](https://github.com/sigi-framework/sigi/compare/v2.1.0...v2.2.0) (2020-06-28)

### Bug Fixes

- **core:** dispatching action on non-effects module causes dead loop ([1ad1100](https://github.com/sigi-framework/sigi/commit/1ad1100158bd131d7366fef165d4acca30e6739b))

# [2.1.0](https://github.com/sigi-framework/sigi/compare/v2.0.2...v2.1.0) (2020-06-02)

**Note:** Version bump only for package @sigi/core
