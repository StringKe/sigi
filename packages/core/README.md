# `@stringke/sigi-core`

## Usage

[Let's start from here](https://sigi.how/en/basic)
