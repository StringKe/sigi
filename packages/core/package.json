{
  "name": "@sigi/core",
  "version": "1.0.0-alpha.4",
  "description": "Sigi core library",
  "keywords": [
    "RxJS",
    "TypeScript"
  ],
  "author": "LongYinan <longyinan.brooklyn@bytedance.com>",
  "homepage": "https://github.com/sigi-framework",
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "esnext": "dist/index.next.js",
  "types": "esm/index.d.ts",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/",
    "access": "public"
  },
  "sideEffects": false,
  "files": [
    "dist",
    "esm"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sigi-framework/sigi.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/sigi-framework/sigi/issues"
  },
  "dependencies": {
    "immer": "^5.3.2"
  }
}
