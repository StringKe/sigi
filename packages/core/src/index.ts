export * from './ayanami'
export * from './types'
export * from './state'
export * from './decorators'
export * from './module.decorator'
export * from './constants'
export * from './logger'
