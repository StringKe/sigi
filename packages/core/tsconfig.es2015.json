{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "target": "ES2015",
    "outDir": "./es2015"
  },
  "references": [{ "path": "../di/tsconfig.next.json" }, { "path": "../types/tsconfig.next.json" }],
  "include": ["src"]
}
