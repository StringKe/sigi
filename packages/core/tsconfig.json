{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./esm",
    "module": "esnext", 
    "composite": true
  },
  "references": [
    { "path": "../di" },
    { "path": "../types" }
  ],
  "include": [
    "src"
  ]
}
