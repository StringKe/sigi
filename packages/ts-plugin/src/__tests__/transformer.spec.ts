import * as ts from 'typescript'
import * as fs from 'fs'
import { resolve } from 'path'

import { SigiTransformer } from '../index'

interface TransformBaseline {
  type: 'transform-baseline'
  filename: string
  content: string
  source: string
  transformed: string
}

expect.addSnapshotSerializer({
  test: (obj: any) => obj && obj.type === 'transform-baseline',
  print: (obj: TransformBaseline, _print: (object: any) => string, indent: (str: string) => string) =>
    `
File: ${obj.filename}
TypeScript before transform:
${indent(obj.content)}
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
${indent(obj.transformed).replace(/ {4}/g, '  ')}
`,
})

describe('ts-plugin specs', () => {
  const fixtureDir = fs.readdirSync(resolve(__dirname, 'fixtures'))
  fixtureDir.forEach((v) => {
    it(`should transform ${v}`, () => {
      const printer = ts.createPrinter()
      const sourceCode = fs.readFileSync(resolve(__dirname, 'fixtures', v), 'utf-8')

      const source = ts.createSourceFile(v, sourceCode, ts.ScriptTarget.ESNext, true)

      const result = ts.transform(source, [SigiTransformer])

      const [transformed] = result.transformed

      const resultCode = printer.printFile(transformed)

      expect({
        transformed: resultCode,
        source: printer.printFile(source),
        filename: `${v}.tsx`,
        type: 'transform-baseline',
        content: sourceCode,
      }).toMatchSnapshot()

      result.dispose()
    })
  })

  it('should throw if Effect option is dynamic', () => {
    const code = `
    import { EffectModule, Module, Effect } from '@sigi/core'
    import { Request } from 'express'
    import { Observable } from 'rxjs'
    import { map } from 'rxjs/operators'
    
    interface AState {}
    
    const effectOption = {
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      }
    }

    @EffectModule('A')
    export class ModuleA extends EffectModule<AState> {
      @Effect(effectOption)
      whatever(payload$: Observable<string>) {
        return payload$.pipe(
          map(() => this.createNoopAction())
        )
      }
    }
    `

    const source = ts.createSourceFile('dynamic-option', code, ts.ScriptTarget.ESNext, true)
    const transpile = () => ts.transform(source, [SigiTransformer])
    expect(transpile).toThrow('Only support object literal parameter in SSREffect')
  })
})
