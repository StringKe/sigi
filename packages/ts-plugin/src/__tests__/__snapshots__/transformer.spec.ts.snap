// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-plugin specs should transform alias 1`] = `

File: alias.tsx
TypeScript before transform:
  import { EffectModule, Module, Effect as _Effect } from '@sigi/core'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @_Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import { EffectModule, Module, Effect as _Effect } from '@sigi/core';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @_Effect({})
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
  }
  

`;

exports[`ts-plugin specs should transform do-nothing 1`] = `

File: do-nothing.tsx
TypeScript before transform:
  import { EffectModule, Module } from '@sigi/core'
  import { Effect } from 'not-sigi'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  
    normalMethod() {
  
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import { EffectModule, Module } from '@sigi/core';
  import { Effect } from 'not-sigi';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello');
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
    normalMethod() {
    }
  }
  

`;

exports[`ts-plugin specs should transform named-import 1`] = `

File: named-import.tsx
TypeScript before transform:
  import { EffectModule, Module, Effect } from '@sigi/core'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({
      skipFirstClientDispatch: true,
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      },
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import { EffectModule, Module, Effect } from '@sigi/core';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({})
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
  }
  

`;

exports[`ts-plugin specs should transform namespace-import 1`] = `

File: namespace-import.tsx
TypeScript before transform:
  import * as Sigi from '@sigi/core'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Sigi.Module('A')
  export class ModuleA extends Sigi.EffectModule<AState> {
    @Sigi.Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import * as Sigi from '@sigi/core';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Sigi.Module('A')
  export class ModuleA extends Sigi.EffectModule<AState> {
    @Sigi.Effect({})
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
  }
  

`;

exports[`ts-plugin specs should transform no-argument 1`] = `

File: no-argument.tsx
TypeScript before transform:
  import { EffectModule, Module, Reducer, Effect } from '@sigi/core'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect()
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  
    @Reducer()
    reducer(state: AState) {
      return { ...state }
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import { EffectModule, Module, Reducer, Effect } from '@sigi/core';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect()
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
    @Reducer()
    reducer(state: AState) {
      return { ...state };
    }
  }
  

`;

exports[`ts-plugin specs should transform without-named-binding 1`] = `

File: without-named-binding.tsx
TypeScript before transform:
  import { EffectModule, Module } from '@sigi/core'
  import { Request } from 'express'
  import { Observable } from 'rxjs'
  import { map } from 'rxjs/operators'
  
  interface AState {}
  
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello')
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(
        map(() => this.createNoopAction())
      )
    }
  }
  
      ↓ ↓ ↓ ↓ ↓ ↓
TypeScript after transform:
  import { EffectModule, Module } from '@sigi/core';
  import { Request } from 'express';
  import { Observable } from 'rxjs';
  import { map } from 'rxjs/operators';
  interface AState {
  }
  @Module('A')
  export class ModuleA extends EffectModule<AState> {
    @Effect({
      payloadGetter: (req: Request) => {
        return require('md5')('hello');
      }
    })
    whatever(payload$: Observable<string>) {
      return payload$.pipe(map(() => this.createNoopAction()));
    }
  }
  

`;
